### PR TITLE
ULTRA merge ultra-rllib into ultra-multiagent-training (cla fix)

### PR DESCRIPTION
**Pull request by AlexLewandowski**
Needed to squash commits to avoid cla issue.

### CODE REVIEW

File: ultra/ultra/baselines/rllib/agent.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                config=config,
+                logger_creator=logger_creator,
+            )
+        # elif self.agent_type=='dqn':
+        #     trainer = dqn.DQNTrainer(
+        #         env=RLlibUltraEnv,
+        #         config=tune_config,
+        #         logger_creator=log_creator(log_dir),
+        #     )
+
```
Comment (christianjans) [inline ultra/ultra/baselines/rllib/agent.py:56]: Should we raise an error if `agent_type` is in `["td3", ..., "sac"]` but is not one of our currently supported trainer types (PPO, DDPG, TD3)? E.g. Should we raise an error if someone wants `agent_type=sac` or `agent_type=dqn`?

File: ultra/ultra/baselines/rllib/agent.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    @staticmethod
+    def rllib_default_config(agent_type):
+        assert agent_type in ["td3", "ddpg", "dqn", "ppo", "sac"]
+
+        if agent_type == "ppo":
+            return ppo.DEFAULT_CONFIG.copy()
+        elif agent_type == "ddpg":
+            return ddpg.DEFAULT_CONFIG.copy()
+        elif agent_type == "td3":
+            return td3.DEFAULT_CONFIG.copy()
```
Comment (christianjans) [inline ultra/ultra/baselines/rllib/agent.py:72]: I guess same thing down here.

File: ultra/tests/test_evaluate.py [hunk 3]
```diff
@@ -315,6 +315,7 @@ def run_experiment(scenario_info, num_agents, log_dir, headless=True):
         agent_id: BaselineAgentSpec(
             action_type=ActionSpaceType.Continuous,
             policy_class=SACPolicy,
+            agent_type="sac",
```
[the remark this answers is not in the document]
Comment (JenishPatel99) [reply ultra/tests/test_evaluate.py:318]: Yes, exactly. If we can used the agent pool to extract the name via the policy then that would be great

File: ultra/ultra/baselines/rllib/agent.py (hunk 1 above)
Comment (JenishPatel99) [reply ultra/ultra/baselines/rllib/agent.py:56]: Hmm, yes that would be needed. I do not know currently if @adreena has managed to get the other policy to work though

**Review by christianjans — APPROVED** [review]

**Review by christianjans — APPROVED** [review]

**Review by JenishPatel99 — APPROVED** [review]

**Pull request merged**